src/Ocelot.Provider.Kubernetes/KubernetesProviderFactory.cs

```diff
@@ -1,41 +1,39 @@
 ﻿using System;
-
+
 using KubeClient;
-
+
 using Microsoft.Extensions.DependencyInjection;
-
+
 using Ocelot.Configuration;
 using Ocelot.Logging;
 using Ocelot.ServiceDiscovery;
+using Ocelot.ServiceDiscovery.Providers;
 
 namespace Ocelot.Provider.Kubernetes
 {
     public static class KubernetesProviderFactory
     {
-        public static ServiceDiscoveryFinderDelegate Get = (provider, config, route) =>
+        private const string PollKube = "pollkube";

```

Comment:
To be honest I don't know, I'm, as you do, amazed by the code smells we can find. I don't understand also, why the colleagues decided to use toLower() for the comparison.

Reply:
Because it is old developer's life hack to compare strings ignoring the case of chars if you don't know the values and cannot apply IgnoreCase comparisons.
Some developers do `string.ToUpper()` 🤣 for both operands to have safe comparing of strings.